### PR TITLE
Added reCAPTCHA support

### PR DESCRIPTION
**Pull request by lyserbin**
With this **huge** patch I added reCAPTCHA support to phpsqlitecms.

You can edit the reCAPTCHA settings under 'Spam protection' section in
the administration panel.

After adding your site and your secret reCAPTCHA keys, you can select
where do you want to add the reCAPTCHA check, by now you can add it to:
* users login (/cms/)
* users comments on commentable posts
* mail sent through formmailer

It also prevents to randomly add Google's script in the page if it isn't
enabled, so it should be good for the ones who don't want Google in
their way ;)

### CODE REVIEW

**Comment by theAsgar** [comment]
Captcha-inventor of the possibilities of modern text recognition are aware and already hindering therefore (even completely unnecessary) human visitors with barely clearly recognizable spelling - and they have the even factored in since they equally provide a link for another Captcha hoping that hidden string might therefrom be guessed correctly.
Blind are hardest hit by the Captcha-delusion. Their access is therefore entirely denied, why captchas on barrier-free pages and may not be used. It should also be clear that there can be no barrier-free captcha. If the string in addition to the Captcha for disabled in other ways - for example, quite simply as an alternative text - would be presented, could be equal to waive the Captcha to the delight of all.
Please no Captcha.

**Comment by lyserbin** [comment]
TL;DR: You are free to not enable it

**Comment by theAsgar** [comment]
I know there are so many beautiful barrier-free alternative without captcha. :)

**Comment by lyserbin** [comment]
You can still not enable it and implement your own way to block bots :sunglasses:
